### ISSUE TITLE
#[instrument(skip_all, fields(x))] silently doesn't work

### ISSUE THREAD

**Issue by safinaskar** [tokio-rs/tracing#2776]
### Version
```text
├── tracing v0.1.40
│   ├── tracing-attributes v0.1.27 (proc-macro)
│   └── tracing-core v0.1.32
└── tracing-subscriber v0.3.17
    ├── tracing-core v0.1.32 (*)
    ├── tracing-log v0.1.3
    │   └── tracing-core v0.1.32 (*)
    └── tracing-serde v0.1.3
        └── tracing-core v0.1.32 (*)
```
### Platform
`Linux comp 5.10.0-0.deb9.24-amd64 #1 SMP Debian 5.10.179-5~deb9u1 (2023-09-01) x86_64 GNU/Linux`

### Crates
tracing, tracing-subscriber

### Description
Consider this code:
```rust
#[tracing::instrument(skip_all, fields(x))]
fn f(x: &str) {
}

fn main() {
    use tracing_subscriber::layer::SubscriberExt;
    use tracing_subscriber::util::SubscriberInitExt;
    use tracing_subscriber::fmt::format::FmtSpan;
    tracing_subscriber::Registry::default()
        .with(tracing_subscriber::fmt::layer()
            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
            .json()
        )
        .init();
    f("a");
}
```
Here is output:
```text
{"timestamp":"2023-10-25T17:50:16.818199Z","level":"INFO","fields":{"message":"new"},"target":"trac","span":{"name":"f"},"spans":[]}
{"timestamp":"2023-10-25T17:50:16.818247Z","level":"INFO","fields":{"message":"close","time.busy":"3.84µs","time.idle":"46.5µs"},"target":"trac","span":{"name":"f"},"spans":[]}
```

As you can see, value for `x` argument is not reported. I. e. `fields(x)` is ignored. But if I replace `fields(x)` with `fields(%x)`, then it works. This is inconsistent. Also, this contradicts well-known Rust principle "If it compiles, then it likely works". So, please, either make `fields(x)` work, either make it compile error

**Comment by kaffarell** [comment]
Hi!
`fields()` is used to add more fields to a span. The exact syntax is explained here: https://docs.rs/tracing/latest/tracing/attr.instrument.html. 
I think what you wanted to do is this:
```rust
#[tracing::instrument(skip_all, fields(x = x))]
fn f(x: &str) {
}
```
What would be your use-case for this?

**Comment by safinaskar** [comment]
> What would be your use-case for this?

I don't have use case for this. I just want to say this: if this syntax is not supported, please, make it compile error

**Comment by davidbarsky** [comment]
We _are_ getting rid of `skip_all` in the master branch. And you are correct: this combination is a little silly, but making it a compile error without bumping the version of `tracing` risks breaking people's builds. Making a semver-incompatible change in a non-semver breaking release is something that requires either an extremely high bar or proof that the impact would be minimal. Sicne we don't have either, I'm inclined to mark this issues as a `wont-fix`. Feel free re-open this, though.

**Comment by safinaskar** [comment]
Thanks! I agree! Also I think the issue should be left opened and marked as "fixed-by-0.2" similarly to how rust project does with its "NLL-fixed-by-NLL" label ( https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3ANLL-fixed-by-NLL ) and other similar labels.

> Feel free re-open this, though

In this repo issues cannot be reopened by non-members
